packages/dashboard/src/components/dashboard.tsx

```diff
@@ -118,32 +131,35 @@ export default function Dashboard(_props: {}): React.ReactElement {
   const trajManager = React.useRef<RobotTrajectoryManager | undefined>(undefined);
   const resourceManager = React.useRef<ResourceManager | undefined>(undefined);
 
+  const { state: _mainMenuState, dispatch: _mainMenuDispatch } = useMainMenuReducer(
+    mainMenuInitialValues,
+  );
+  const mainMenuState = React.useMemo(() => _mainMenuState, [_mainMenuState]);
+  const {
+    currentView,
+    settings,
+    showOmniPanel,
+    stackNavigator,
+    showSettings,
+    showHelp,
+    showHotkeysDialog,
+  } = mainMenuState;
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const mainMenuDispatch = React.useMemo(() => _mainMenuDispatch, []);
```

Comment:
It might mean bad design if we have to disable eslint checks. Maybe you can try

1. return the dispatch directly
2. move `useMemo` inside `useMainMenuReducer`. Add dispatch to the deps, or add the eslint ignore there (with a comment explaining it) so that callers don't have to do this.

Reply:
Thanks! Yeah, I move them to the custom hook `useMainMenuReducer` I managed to memoize the state and the dispatch there. Looks cleaner now.